.eslintrc.js

```diff
@@ -1,46 +1,299 @@
-// This is a convenience field for what rules we want to use
-const rules = {
-  'semi': ['error', 'always'],
+/**
+ * This file represents the best approximation of our development conventions in this project as of
+ * the conversion from TSLint to ESLint in API-1323. The rules in this file should be considered
+ * flexible and can be extended or relaxed as necessary. Feel free to add rules that would be helpful
+ * and remove rules that are overly strict. As of API-3321, these rules were added to the backend to
+ * keep the linting in line with the frontend. When adding or changing a rule here, you should ask
+ * if it makes sense to update or add the rule in the frontend as well. Please try to keep the two
+ * repos in sync when possible.
+ *
+ * Notes
+ *  1. The Typescript ESLint plugin has several rules that extend rules in the core ESLint module. In
+ *  general, we prefer to use the Typescript ESLint version. You can see a list of the extended rules
+ *  here: https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules
+ *  2. Some rules can have a higher cost performance than others. To test performance of the rule set,
+ *  run "TIMING=1 npm run lint". The TIMING flag causes ESLint to print out metrics on time to process
+ *  each rule. https://eslint.org/docs/developer-guide/working-with-rules#per-rule-performance
+ *
+ * Presets included
+ *  - Typescript ESLint: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.ts
+ *    - ESLint via Typescript ESLint: https://github.com/eslint/eslint/blob/master/conf/eslint-recommended.js
+ *  - Typescript ESLint with type checking: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended-requiring-type-checking.ts
+ */
+
+// https://eslint.org/docs/rules/
+const coreESLintRules = {
+  'accessor-pairs': 'error',
+  'array-bracket-spacing': 'error',
+  'array-callback-return': 'error',
+  'array-element-newline': ['error', 'consistent'],
+  'arrow-body-style': 'error',
+  'arrow-parens': ['error', 'as-needed'],
+  'arrow-spacing': 'error',
+  'block-scoped-var': 'error',
+  'block-spacing': 'error',
+  'class-methods-use-this': 'error',
+  'comma-style': 'error',
+  complexity: ['error', 12],
+  'computed-property-spacing': 'error',
+  'consistent-return': 'error',
+  curly: 'error',
+  'default-case': 'error',
+  'default-case-last': 'error',
+  'dot-location': ['error', 'property'],
+  'eol-last': 'error',
+  eqeqeq: ['error', 'always', { null: 'ignore' }],
+  'grouped-accessor-pairs': 'error',
+  'guard-for-in': 'error',
+  'id-denylist': [
+    'error',
+    'any',
+    'Number',
+    'number',
+    'String',
+    'string',
+    'Boolean',
+    'boolean',
+    'Undefined',
+    'undefined',
+  ],
+  'id-length': [
+    'error',
+    {
+      min: 1,
+      max: 32,
+    },
+  ],
+  'jsx-quotes': 'error',
```

Comment:
we can probably take out the react specific stuff

Reply:
good catch